arch/arm/src/stm32wb/stm32wb_rtc.c

```diff
@@ -0,0 +1,1831 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdbool.h>
+#include <errno.h>
+#include <debug.h>
+#include <assert.h>
+
+#include <nuttx/arch.h>
+#include <nuttx/irq.h>
+#include <nuttx/time.h>
+#include <arch/board/board.h>
+
+#include "arm_internal.h"
+#include "chip.h"
+#include "stm32wb_rcc.h"
+#include "stm32wb_pwr.h"
+#include "stm32wb_rtc.h"
+#include "stm32wb_exti.h"
+
+#ifdef CONFIG_STM32WB_RTC
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Configuration ************************************************************/
+
+/* This RTC implementation supports
+ *  - date/time RTC hardware
+ *  - extended functions Alarm A and B
+ * */
+
+#ifndef CONFIG_RTC_DATETIME
+#  error "CONFIG_RTC_DATETIME must be set to use this driver"
+#endif
+
+#ifdef CONFIG_RTC_HIRES
+#  error "CONFIG_RTC_HIRES must NOT be set with this driver"
+#endif
+
+#ifndef CONFIG_STM32WB_PWR
+#  error "CONFIG_STM32WB_PWR must selected to use this driver"
+#endif
+
+/* Constants ****************************************************************/
+
+#define SYNCHRO_TIMEOUT               (0x00020000)
+#define INITMODE_TIMEOUT              (0x00010000)
+
+#define RTC_ALRMR_ENABLE              (0x00000000)
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+typedef unsigned int rtc_alarmreg_t;
+
+struct alm_cbinfo_s
+{
+  volatile alm_callback_t ac_cb; /* Client callback function */
+  volatile void *ac_arg;         /* Argument to pass with the callback function */
+};
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+/* Callback to use when an EXTI is activated  */
+
+static struct alm_cbinfo_s g_alarmcb[RTC_ALARM_LAST];
+static bool g_alarm_enabled;  /* True: Alarm interrupts are enabled */
+#endif
+
+#ifdef CONFIG_RTC_PERIODIC
+static wakeupcb_t g_wakeupcb;
+static bool g_wakeup_enabled;  /* True: Wakeup interrupts are enabled */
+#endif
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/* g_rtc_enabled is set true after the RTC has successfully initialized */
+
+volatile bool g_rtc_enabled = false;
```

Comment:
Optional:
```suggestion
volatile bool g_rtc_enabled;
```
zero inited according to C standard